File: csharp/SecasignTest/sdk/authentication/AuthenticationTest.cs
using Secasign;
using Secasign.dto.authentication;
using SecasignTest.extension;

namespace SecasignTest.sdk.authentication;

public class AuthenticationTest
{
    private static SdkConfiguration SdkConfiguration => SdkExtension.SdkConfiguration;

    [SdkExtension(SdkTestRequirements.IsLoginAvailable)]
    public async Task TestAuthentication()
    {
        // Create the HTTP client
        var secasignHttpClient = new SecasignHttpClient(SdkConfiguration.Url!);

        // Login and obtain a JWT token
        var authenticationDto = new AuthenticationDto(SdkConfiguration.EmailAddress!, SdkConfiguration.Password!);
        var tokenDto = await secasignHttpClient.Login(authenticationDto);
        Assert.NotNull(tokenDto.token);
    }
}

File: java/src/test/java/com/secacon/secasign/sdk/authentication/AuthenticationTest.java
package com.secacon.secasign.sdk.authentication;

import com.secacon.secasign.extension.SdkConfiguration;
import com.secacon.secasign.extension.SdkTest;
import com.secacon.secasign.extension.SdkTestRequirements;
import com.secacon.secasign.sdk.SecasignHttpClient;
import com.secacon.secasign.sdk.dto.authentication.AuthenticationDto;
import com.secacon.secasign.sdk.dto.authentication.TokenDto;

import static org.junit.jupiter.api.Assertions.assertNotNull;

/**
 * Test a simple login.
 *
 * @author Simon Wächter
 */
@SdkTest
public class AuthenticationTest {

    /**
     * Test the login.
     *
     * @param sdkConfiguration SDK configuration
     * @throws Exception Exception in case of a problem
     */
    @SdkTestRequirements(isLoginAvailable = true)
    public void testAuthentication(SdkConfiguration sdkConfiguration) throws Exception {
        // Create the HTTP client
        SecasignHttpClient secasignHttpClient = new SecasignHttpClient(sdkConfiguration.getUrl());

        // Obtain the token
        TokenDto tokenDto = secasignHttpClient.login(new AuthenticationDto(sdkConfiguration.getEmailAddress(), sdkConfiguration.getPassword()));
        assertNotNull(tokenDto.getToken());
    }
}
